src/main/java/com/redhat/microprofile/psi/internal/quarkus/core/properties/QuarkusConfigMappingProvider.java

```diff
@@ -145,19 +146,23 @@ private void populateConfigObject(PsiClass configMappingType, String prefixStr,
 				PsiType psiType = method.getReturnType();
 				String returnTypeSignature = getResolvedResultTypeName(method);
 				String resolvedTypeSignature = getRawResolvedTypeName(method);
-				if (isOptional(resolvedTypeSignature)) {
+				if (isOptional(psiType)) {
 					// it's an optional type
 					// Optional<List<String>> databases();
 					// extract the type List<String>
-					returnTypeSignature = getResolvedTypeName(((PsiClassType) psiType).getParameters()[0]);
-					resolvedTypeSignature = getRawResolvedTypeName(((PsiClassType) psiType).getParameters()[0]);
-					psiType = ((PsiClassType) psiType).getParameters()[0];
+					psiType = getTypeParameter(psiType);
+					if (psiType != null) {
+						resolvedTypeSignature = getRawResolvedTypeName(psiType);
+					}
 				}
 
 				PsiClass returnType = findType(method.getManager(), resolvedTypeSignature);
 				boolean simpleType = isSimpleType(resolvedTypeSignature, returnType);
+
 				if (!simpleType) {
-					if (returnType != null && !returnType.isInterface()) {
+					if (returnType != null
+							&& !returnType.isInterface()
+							&& !isInJavaPackage(returnType)) {
```

Comment:
Please remove isInJavaPackage I find it is not clear, You said me that tests fails if you modify isSimpleType with java.util.OptionalInt, could you tell me which test is failing?